### PR TITLE
feat: Replace deprecated is_enabled with state in EventBridge rules

### PR DESCRIPTION
**Pull request by hammy414**
## Description
Updated the aws_cloudwatch_event_rule resource definition in the Terraform configuration to replace the deprecated is_enabled attribute with the state attribute. This change ensures compatibility with the latest versions of the AWS provider.

## Motivation and Context
This change is required to address deprecation warnings from the AWS provider for Terraform. The use of is_enabled has been deprecated and replaced by state, which is now required to define the enabled or disabled status of CloudWatch Event Rules.
Resolving this prevents potential issues with future versions of the AWS provider that may no longer support the is_enabled attribute, ensuring the stability and maintainability of the infrastructure code.

## Breaking Changes
This change does not break backwards compatibility with the current major version of the module, as it merely replaces a deprecated attribute with its recommended alternative.
Users who rely on older versions of the AWS provider (before the introduction of the state attribute) should consider upgrading their provider to continue using this module.

## How Has This Been Tested?
- [X] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [X] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [X] I have executed `pre-commit run -a` on my pull request
- ❯ pre-commit run -a
[INFO] Initializing environment for https://github.com/antonbabenko/pre-commit-terraform.
[INFO] Installing environment for https://github.com/pre-commit/pre-commit-hooks.
[INFO] Once installed this environment will be reused.
[INFO] This may take a few minutes...
Terraform fmt............................................................Passed
Terraform docs...........................................................Passed
Terraform validate with tflint...........................................Passed
Terraform validate.......................................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed

### CODE REVIEW

**Comment by T0tt1** [comment]
+1 for moving fw with this PR.

**Comment by T0tt1** [comment]
@antonbabenko this looks like quick fix and easy to adopt in the module. It will allow all of use to stop be annoyed by the constant warnings during TF plan/apply.

File: main.tf [hunk 1]
```diff
@@ -88,11 +88,10 @@ resource "aws_cloudwatch_event_rule" "this" {
   event_bus_name = var.create_bus ? aws_cloudwatch_event_bus.this[0].name : var.bus_name
 
   description         = lookup(each.value, "description", null)
-  is_enabled          = lookup(each.value, "enabled", null)
   event_pattern       = lookup(each.value, "event_pattern", null)
   schedule_expression = lookup(each.value, "schedule_expression", null)
   role_arn            = lookup(each.value, "role_arn", false) ? aws_iam_role.eventbridge[0].arn : null
-  state               = lookup(each.value, "state", null)
+  state               = coalesce(lookup(each.value, "enabled", null), false) ? "ENABLED" : "DISABLED"
```
Comment (antonbabenko) [inline main.tf:94]:
```suggestion
  state               = try(each.value.enabled ? "ENABLED" : "DISABLED", tobool(each.value.state) ? "ENABLED" : "DISABLED", "DISABLED")
```
How about this? It should handle previous cases (with "enabled" and "state") and be backward-compatible.

**Review by antonbabenko — CHANGES_REQUESTED** [review]

File: main.tf (hunk 1 above)
Comment (antonbabenko) [reply main.tf:94]: Will this work?

Comment (hammy414) [reply main.tf:94]: Hey @antonbabenko Yes that works, and deploys fine. 

However, I now get the following error when running the prehook:

❯ pre-commit run -a
Terraform fmt............................................................Passed
Terraform docs...........................................................Failed
- hook id: terraform_docs
- files were modified by this hook
Terraform validate with tflint...........................................Passed
Terraform validate.......................................................Passed
check for merge conflicts................................................Passed
fix end of files.........................................................Passed
trim trailing whitespace.................................................Passed

File: main.tf [hunk 2]
```diff
@@ -88,11 +88,10 @@ resource "aws_cloudwatch_event_rule" "this" {
   event_bus_name = var.create_bus ? aws_cloudwatch_event_bus.this[0].name : var.bus_name
 
   description         = lookup(each.value, "description", null)
-  is_enabled          = lookup(each.value, "enabled", null)
   event_pattern       = lookup(each.value, "event_pattern", null)
   schedule_expression = lookup(each.value, "schedule_expression", null)
   role_arn            = lookup(each.value, "role_arn", false) ? aws_iam_role.eventbridge[0].arn : null
-  state               = lookup(each.value, "state", null)
+  state               = try(each.value.enabled ? "ENABLED" : "DISABLED", tobool(each.value.state) ? "ENABLED" : "DISABLED", "DISABLED")
```
Comment (antonbabenko) [inline main.tf:94]:
```suggestion
  state               = try(each.value.enabled ? "ENABLED" : "DISABLED", tobool(each.value.state) ? "ENABLED" : "DISABLED", upper(each.value.state), "DISABLED")
```

**Review by antonbabenko — APPROVED** [review]

**Comment by antonbabenko** [comment]
This PR is included in version 3.3.0 :tada:

**Comment by T0tt1** [comment]
Great news! Thank you for prompt response.

**Pull request merged**